Sources/ProcedureKit/Support.swift

```diff
@@ -113,6 +113,107 @@ public extension Protector where T: RangeReplaceableCollection {
             ward.append(contentsOf: newElements)
         }
     }
+
+    func append<C : Collection>(contentsOf newElements: C) where C.Iterator.Element == T.Iterator.Element {
+        write { (ward: inout T) in
+            ward.append(contentsOf: newElements)
+        }
+    }
+}
+
+
+/// A wrapper class for a pthread_mutex
+final public class PThreadMutex {
+    private var mutex = pthread_mutex_t()
+
+    public init() {
+        let result = pthread_mutex_init(&mutex, nil)
+        precondition(result == 0, "Failed to create pthread mutex")
+    }
+
+    deinit {
+        let result = pthread_mutex_destroy(&mutex)
+        assert(result == 0, "Failed to destroy mutex")
+    }
+
+    fileprivate func lock() {
+        let result = pthread_mutex_lock(&mutex)
+        assert(result == 0, "Failed to lock mutex")
+    }
+
+    fileprivate func unlock() {
+        let result = pthread_mutex_unlock(&mutex)
+        assert(result == 0, "Failed to unlock mutex")
+    }
+
+    /// Convenience API to execute block after acquiring the lock
+    ///
+    /// - Parameter block: the block to run
+    /// - Returns: returns the return value of the block
+    public func withCriticalScope<T>(block: () -> T) -> T {
+        lock()
+        defer { unlock() }
+        let value = block()
+        return value
+    }
+}
+
+public class MutexProtector<T> {
```

Comment:
Is this used instead of `Protector`? We could just name this `Protector` and remove the previous version?